packages/chevrotain/src/scan/lexer_public.ts

```diff
@@ -523,8 +524,11 @@ export class Lexer {
     push_mode.call(this, initialMode)
 
     let currConfig!: IPatternConfig
+    let failedBeforeEnd = false
 
-    while (offset < orgLength) {
+    const lexerConfig = this.config
```

Comment:
Minor: good, but I meant something more akin to:
```suggestion
    const recoveryEnabled = this.config.recoveryEnabled
```